File: baekjoon/dp/2748.py
# 피보나치

def fibo(n):
    if n==0:
        return 0
    if n==1:
        return 1
    return fibo(n-1) + fibo(n-2)

n = int(input())
def fibo_dp(n):
    ans = [0]*(n+1)
    ans[1] = 1
    for i in range(2,n+1):
        ans[i] = ans[i-1] + ans[i-2]
    return ans[n]
print(fibo_dp(n))

File: baekjoon/dp/2748.c
// 피보나치

#include <stdio.h>

// int fibo(int n) {
//     if (n==0) {return 0;}
//     else if (n==1) {return 1;}
//     else {return fibo(n-1) + fibo(n-2);}
// }


long long int ans[91];
long long int fibo_dp(int n) {
    ans[0] = 0;
    ans[1] = 1;
    if (n>1) {
        for (int i=2; i<=n; i++) {
            ans[i] = ans[i-1] + ans[i-2];
        }
    }
    return ans[n];
}


int main() {

    int n;
    scanf("%d", &n);
    printf("%lld\n", fibo_dp(n));

    return 0;
}